client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/AbstractKaaClient.java

```diff
@@ -205,12 +205,14 @@
         transportContext.getUserTransport().setEndpointRegistrationProcessor(endpointRegistrationManager);
         transportContext.getLogTransport().setLogProcessor(logCollector);
         transportContext.initTransports(this.channelManager, this.kaaClientState);
-        
+
         eventFamilyFactory = new EventFamilyFactory(eventManager, context.getExecutorContext());
     }
 
     @Override
-    public void start() {
+    public void start() throws KaaException {
```

Comment:
I suggest to use both ways. If a listener is specified, notify it, otherwise, throw an exception.